palace/fem/qfunctions/33/hcurlhdiv_error_33_qf.h

```diff
@@ -0,0 +1,82 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef PALACE_LIBCEED_HCURLHDIV_ERROR_33_QF_H
+#define PALACE_LIBCEED_HCURLHDIV_ERROR_33_QF_H
+
+#include "../coeff/coeff_3_qf.h"
+#include "utils_33_qf.h"
+
+CEED_QFUNCTION(f_apply_hcurlhdiv_error_33)(void *__restrict__ ctx, CeedInt Q,
+                                           const CeedScalar *const *in,
+                                           CeedScalar *const *out)
+{
+  const CeedScalar *attr = in[0], *wdetJ = in[0] + Q, *adjJt = in[0] + 2 * Q, *u1 = in[1],
+                   *u2 = in[2];
+  CeedScalar *v = out[0];
+
+  CeedPragmaSIMD for (CeedInt i = 0; i < Q; i++)
+  {
+    CeedScalar adjJt_loc[9], v1_loc[3], v2_loc[3];
+    MatUnpack33(adjJt + i, Q, adjJt_loc);
+    {
+      const CeedScalar u1_loc[3] = {u1[i + Q * 0], u1[i + Q * 1], u1[i + Q * 2]};
+      CeedScalar coeff[6];
+      CoeffUnpack3((const CeedIntScalar *)ctx, (CeedInt)attr[i], coeff);
+      MultBAx33(adjJt_loc, coeff, u1_loc, v1_loc);
+    }
+    {
+      const CeedScalar u2_loc[3] = {u2[i + Q * 0], u2[i + Q * 1], u2[i + Q * 2]};
+      CeedScalar coeff[6], J_loc[9];
```

Comment:
By "hoisted" do you mean put into the outer scope? I would think any compiler would not stack allocate two `coeff[6]` objects here since only one is in scope at a time. Is that not true? I hadn't thought about it but certainly moving it out could reduce local stack/register usage if that's not true.

Reply:
I think a sane compiler would notice, you're right with the scope. I was being dumb and thinking of a heap variable. Should be fine as is.